Makefile

```diff
@@ -61,7 +63,15 @@ clean:
 	$(GOCLEAN)
 	rm -f $(PROGRAM)
 
+# Download the installation script for golangci-lint, verify its SHA-256 digest,
+# run it if everything checks out, and verify the resulting binary.
 $(GOLANGCI_LINT):
-	curl -sfL \
-			https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
-		| sh -s -- -b $(GOBIN) $(GOLANGCI_LINT_VERSION)
+	mkdir -p $(dir $@)
```

Comment:
I am trying to wrap my head around what this line does.
`$@`: the name of the target being generated
So `$@` in this case is `$(GOLANGCI_LINT)`, which expands to `$(CURDIR)/bin/golangci-lint`
And I get the following when I just try the `dir` command:
```
dir $PWD/bin/golangci-lint
dir: cannot access '/home/jonasal/bin/golangci-lint': No such file or directory
```
while I get all the filenames and folders present in the case the path actually exist.

Is it something I am missing with this or will this fail in the ci case when the `bin/` folder has not been created?

Reply:
`$(dir ...)` isn't a call to a shell command but a call to a [make function](https://www.gnu.org/software/make/manual/html_node/File-Name-Functions.html#index-dir) that's basically equivalent to dirname(1). IOW, line 69 will expand to `mkdir -p $(CURDIR)/bin/`. If you want your command to actually contain a dollar sign for e.g. a shell variable reference you need `$$`. Have I ever forgotten this and been confused over my makefile not working to my satisfaction? Yes. Yes, I have.

Reply:
Alrighty! :)